tests/components/roku/test_init.py

```diff
@@ -0,0 +1,68 @@
+"""Tests for the Roku integration."""
+from socket import gaierror as SocketGIAError
+
+from asynctest import patch
+from requests.exceptions import RequestException
+from roku import RokuException
+
+from homeassistant.components.roku.const import DOMAIN
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    ENTRY_STATE_SETUP_RETRY,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from tests.components.roku import MockDeviceInfo, setup_integration
+
+
+async def test_config_entry_not_ready(hass: HomeAssistantType) -> None:
+    """Test the Roku configuration entry not ready."""
+    with patch(
+        "homeassistant.components.roku.Roku._call", side_effect=RokuException,
+    ):
+        entry = await setup_integration(hass)
+
+    assert entry.state == ENTRY_STATE_SETUP_RETRY
+
+
+async def test_config_entry_not_ready_request(hass: HomeAssistantType) -> None:
+    """Test the Roku configuration entry not ready."""
+    with patch(
+        "homeassistant.components.roku.Roku._call", side_effect=RequestException,
+    ):
+        entry = await setup_integration(hass)
+
+    assert entry.state == ENTRY_STATE_SETUP_RETRY
+
+
+async def test_config_entry_not_ready_socket(hass: HomeAssistantType) -> None:
+    """Test the Roku configuration entry not ready."""
+    with patch(
+        "homeassistant.components.roku.Roku._call", side_effect=SocketGIAError,
+    ):
+        entry = await setup_integration(hass)
+
+    assert entry.state == ENTRY_STATE_SETUP_RETRY
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184723)